src/prefect/client/client.py

```diff
@@ -640,8 +644,42 @@ def register(
             if compressed
             else res.data.createFlow.id
         )
+
+        if not flow_id_only:
+            # Generate direct link to Cloud flow
+            tenant_slug = self.get_default_tenant_slug()
+
+            url = (
+                re.sub("api-", "", prefect.config.cloud.api)
+                if re.search("api-", prefect.config.cloud.api)
+                else re.sub("api", "cloud", prefect.config.cloud.api)
+            )
+
+            def join(url: str, items: list) -> str:
+                if len(items) == 1:
+                    return urljoin(url, items[0])
+                return join(urljoin(url, items[0]) + "/", items[1:])
+
+            flow_url = join(url, [tenant_slug, "flow", flow_id])
+
+            print("Flow: {}".format(flow_url))
+
         return flow_id
 
+    def get_default_tenant_slug(self) -> str:
+        """
+        Get the default tenant slug for the currently authenticated user
+
+        Returns:
+            - str: the slug of the current default tenant for this user
+        """
+        res = self.graphql(
+            query={"query": {"user": {"default_membership": {"tenant": "slug"}}}}
```

Comment:
Question: why are you using the default value?  E.g., I could imagine someone registering a flow off a token that doesn't correspond to their default membership.

Reply:
I was working with the idea that whichever your current tenant is in conjunction with the token used to authenticate the client would be the ideal tenant to use. I could add an option to also provide a tenant slug if you think it would be useful.